### PR TITLE
Mplus 2: Version 1.000; Mplus 1: Version 1.000

### PR DESCRIPTION
**Pull request by aaronbell**
438bea5: [gftools-packager] Mplus 1: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 1 Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/8c6c65ff55f1725b9693b4c3f11d4c253fce8dd0.

 c5ae89a: [gftools-packager] Mplus 2: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 2 Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/8c6c65ff55f1725b9693b4c3f11d4c253fce8dd0.

 e97a9da: [gftools-packager] Mplus 1 Code: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 1 Code Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/8c6c65ff55f1725b9693b4c3f11d4c253fce8dd0.

 a29c278: [gftools-packager] Mplus Code Latin: Version 1.000; ttfautohint (v1.8.3) added

* Mplus Code Latin Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/8c6c65ff55f1725b9693b4c3f11d4c253fce8dd0.

### CODE REVIEW

**Comment by aaronbell** [comment]
FYI @mintoming

**Comment by aaronbell** [comment]
Regarding the FAILs on MplusCodeLatin:
- Font contains a width axis, and per the spec, it has been left without the widths being included in the instance names, causing this error. 
- Font metrics have been left to match the other Mplus CJK fonts. This way, if it is used in conjunction with any of the other ones, it will not cause and metrics issues. 
- Font is not monospace :)

**Comment by aaronbell** [comment]
Thomas Linard has requested the addition of ExtraLight and SemiBold to the variable font STAT table:
https://github.com/coz-m/MPLUS_FONTS/issues/14

We are working to resolve upstream.

**Comment by aaronbell** [comment]
Updated

## Mplus Code Latin: Version 1.000; ttfautohint (v1.8.3) added; Mplus 1 Code: Version 1.000; ttfautohint (v1.8.3) added; Mplus 1: Version 1.000; ttfautohint (v1.8.3) added; Mplus 2: Version 1.000; ttfautohint (v1.8.3) added

---
 25bb9a5: [gftools-packager] Mplus 1: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 1 Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/3d3f039069da6b3dc8720c66b87b533481930e15.

 68194ea: [gftools-packager] Mplus 1 Code: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 1 Code Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/3d3f039069da6b3dc8720c66b87b533481930e15.

 acfed76: [gftools-packager] Mplus 2: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 2 Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/3d3f039069da6b3dc8720c66b87b533481930e15.

 c303d0d: [gftools-packager] Mplus Code Latin: Version 1.000; ttfautohint (v1.8.3) added

* Mplus Code Latin Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/3d3f039069da6b3dc8720c66b87b533481930e15.

**Comment by aaronbell** [comment]
Now updated with two extra weights in each family

**Comment by aaronbell** [comment]
I've dehinted the fonts with Kanji to save space.

File: ofl/mplus1/METADATA.pb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: "Mplus 1"
+designer: "MPlus Fonts"
```
Comment (davelab6) [inline ofl/mplus1/METADATA.pb:2]: This should match the attribution on the existing M PLUS fonts (eg https://fonts.google.com/specimen/M+PLUS+Rounded+1c)

File: ofl/mplus1/METADATA.pb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: "Mplus 1"
```
Comment (davelab6) [inline ofl/mplus1/METADATA.pb:1]: This should match the naming used already, eg https://fonts.google.com/specimen/M+PLUS+Rounded+1c

**Review by davelab6 — CHANGES_REQUESTED** [review]
Looking forwards to this :) However, the few minor things I commented on will, I guess, require a rebuild and coordination with the upstream to propagate them back there, so please work with @mintoming on this :)

File: ofl/mplus1/METADATA.pb (hunk 2 above)
Comment (aaronbell) [reply ofl/mplus1/METADATA.pb:1]: Ah! For some reason I hadn't realized there was an extant MPLUS. I'll match it.

**Comment by aaronbell** [comment]
@davelab6 These appear to be primarily metadata.pb changes rather than modifications to the actual font files. Internally, M PLUS 1P uses the "Mplus1p" name:

Screen Shot 2021-06-15 at 8 33 29 AM

I do need to rebuild the Code version as Coji's updated it, but aside from that, I don't think we need to rebuild the others.

**Comment by aaronbell** [comment]
Updated

## M PLUS 1 Code: Version 1.000 added; M PLUS 2: Version 1.000 added; M PLUS 1: Version 1.000 added; M PLUS Code Latin: Version 1.000; ttfautohint (v1.8.3) added

---
 bd10172: [gftools-packager] M PLUS 1: Version 1.000 added

* M PLUS 1 Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 7b805a9: [gftools-packager] M PLUS 1 Code: Version 1.000 added

* M PLUS 1 Code Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 bfc48c4: [gftools-packager] M PLUS 2: Version 1.000 added

* M PLUS 2 Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 304e509: [gftools-packager] M PLUS Code Latin: Version 1.000; ttfautohint (v1.8.3) added

* M PLUS Code Latin Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

**Comment by aaronbell** [comment]
Updated

## M PLUS 1: Version 1.000 added; M PLUS 2: Version 1.000 added; M PLUS 1 Code: Version 1.000 added; M PLUS Code Latin: Version 1.000; ttfautohint (v1.8.3) added

---
 bd10172: [gftools-packager] M PLUS 1: Version 1.000 added

* M PLUS 1 Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 7b805a9: [gftools-packager] M PLUS 1 Code: Version 1.000 added

* M PLUS 1 Code Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 bfc48c4: [gftools-packager] M PLUS 2: Version 1.000 added

* M PLUS 2 Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 304e509: [gftools-packager] M PLUS Code Latin: Version 1.000; ttfautohint (v1.8.3) added

* M PLUS Code Latin Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

**Comment by aaronbell** [comment]
Had some issues with conflicts between my local version of this and the repro :/. Think I got it all sorted out. Anyway, ended up rebuilding all the fonts to implement the fix Coji requested, and have updated metadata so this should be good to go now.

**Comment by RosaWagner** [comment]
There are intriguing FAILs about duplicated instances and naming inconsistencies.

I wonder if we shouldn't update the v-metrics of the already published fonts, instead of adding new bad ones

**Comment by aaronbell** [comment]
I can give a bit more clarity about the FAILS on the MPlus Latin font. 

1) It contains a width axis, which is what is leading to the "duplicated instances" (since GF doesn't allow for the width axis to be present in the fvar table). 

2) All of these are following the CJK vertical metrics guidelines as they're part of the same family. So while font bakery calls these FAILS, they're OK as is. 

3) I'll check the naming inconsistencies. I recall there was some confusion as MPlus was already on GF, so I needed to snap to the naming convention already established. Might have missed these.

**Comment by RosaWagner** [comment]
>It contains a width axis, which is what is leading to the "duplicated instances" (since GF doesn't allow for the width axis to be present in the fvar table).

Indeed, the width=125 instances shouldn't be in the FVAR at all

**Comment by davelab6** [comment]
The `ofl/mplus1/static/` folder and its contents do not need to exist here, since they are not hand hinted.

**Comment by davelab6** [comment]
> Font metrics have been left to match the other Mplus CJK fonts. This way, if it is used in conjunction with any of the other ones, it will not cause and metrics issues.

I discussed this with @RosaWagner and @m4rc1e and we agreed this makes sense. 

> @davelab6 These appear to be primarily metadata.pb changes rather than modifications to the actual font files. Internally, M PLUS 1P uses the "Mplus1p" name:

Thanks for fixing this throughout, I just hacked it quickly in 2016 and didn't change the files internals.

**Comment by RosaWagner** [comment]
@aaronbell I still don't have a working gftools[qa], can you send me a zip ?
also, should the 1p font be deprecated after this goes to production?

**Comment by aaronbell** [comment]
1P is a rounded version which is different than these, so it shouldn't be deprecated. That said, it would probably be worth investigating if a new version of 1P should also be produced. 

For QA, are you just looking for fontbakery? Or something else?

**Comment by RosaWagner** [comment]
Also browser previews; I can still generate fontbakery report, but I can't have browser screenshots to check consistency across platform and eventual visual weirdness.

**Pull request merged**